command/agent/agent.go

```diff
@@ -458,6 +458,15 @@ func (a *Agent) RPC(method string, args interface{}, reply interface{}) error {
 	return a.client.RPC(method, args, reply)
 }
 
+// RPCRetry is used to make an RPC call to the Consul servers.  This allows
+// the agent to implement the Consul.Interface.
+func (a *Agent) RPCRetry(method string, args interface{}, reply interface{}, timeout time.Duration) error {
```

Comment:
I thought about that when writing the interface and settled back on duration because the `Agent.RPC*()` interface is where the caller gets to specify their contract.  It's six of one, half dozen of the other but specifying the interface in terms of the contract language ("you have 500ms to complete this action") seemed more explicit and natural than an absolute point in time ("finish this action by 09:46:00.500").  Use of timeout durations fit better conceptually than deadlines.  I'm okay changing this or adding a new wrapper `RPCDeadline()`.